CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs

```diff
@@ -0,0 +1,142 @@
+    /// them to standard out
+    /// </summary>
+    public class TopNWordsTask
+    {
+        public static void TaskMain(string[] args)
+        {
+            TelemetryClient insightsClient = new TelemetryClient()
+            {
+                InstrumentationKey = "YOUR-IKEY-GOES-HERE",
+            };
+
+            if (args == null || args.Length != 5)
+            {
+                Exception e = new Exception("Usage: TopNWordsSample.exe --Task <blobpath> <numtopwords> <storageAccountName> <storageAccountKey>");
+                throw e;
+            }
+
+            string blobName = args[1];
+            int numTopN = int.Parse(args[2]);
+            string storageAccountName = args[3];
+            string storageAccountKey = args[4];
+
+            WordCount wordCounter = new WordCount(insightsClient);
+            wordCounter.CountWords(blobName, numTopN, storageAccountName, storageAccountKey);
+            wordCounter.Dispose();
+        }
+    }
+
+    public class WordCount
+    {
+        public Dictionary<string, string> CommonProperties
+        {
+            get
+            {
+                if (this.commonProperties == null)
+                {
+                    this.commonProperties = new Dictionary<string, string>();
+                    this.commonProperties.Add("JobId", Environment.GetEnvironmentVariable("AZ_BATCH_JOB_ID"));
+                    this.commonProperties.Add("TaskId", Environment.GetEnvironmentVariable("AZ_BATCH_TASK_ID"));
+                    this.commonProperties.Add("PoolId", Environment.GetEnvironmentVariable("AZ_BATCH_POOL_ID"));
+                    this.commonProperties.Add("NodeId", Environment.GetEnvironmentVariable("AZ_BATCH_NODE_ID"));
+                }
+
+                return this.commonProperties;
+            }
+        }
+
+        private TelemetryClient insightsClient;
+        private Dictionary<string, string> commonProperties;
+        private string taskId;
+        bool disposed = false;
+
+        public WordCount(TelemetryClient telemetryClient)
+        {
+            this.insightsClient = telemetryClient;
+            this.taskId = Environment.GetEnvironmentVariable("AZ_BATCH_TASK_ID");
+        }
+
+        public void CountWords(string blobName, int numTopN, string storageAccountName, string storageAccountKey)
+        {
+            // simulate exception
+            Random rand = new Random();
+            if (rand.Next(0, 10) % 10 == 0)
+            {
+                blobName += ".badUrl";
+            }
+
+            // open the cloud blob that contains the book
+            var storageCred = new StorageCredentials(storageAccountName, storageAccountKey);
+            CloudBlockBlob blob = new CloudBlockBlob(new Uri(blobName), storageCred);
+            using (Stream memoryStream = new MemoryStream())
+            {
+                // Find blob download time
+                DateTime start = DateTime.Now;
+                blob.DownloadToStream(memoryStream);
```

Comment:
No reason. Since the files are small, there really isn't any use to download it to disk just to load it up into memory again.